src/Components/Facility/FacilityFilter/index.tsx

```diff
@@ -59,6 +61,11 @@ function FacilityFilter(props: any) {
   useAbortableEffect((status: statusType) => {
     fetchStates(status);
   }, []);
+  useEffect(() => {
+    if (!isPreventChangeFilterStateOn) {
+      setFilterState(filter);
+    }
+  }, [filter]);
```

Comment:
not a fan of this hack.

Reply:
Why would you need that to be prevented?

Reply:
@rithviknishad There are many state changes happening to the filterState . So I want the changes to be synced only when someone clicks on apply filter button.